src/mapeo-manager.js

```diff
@@ -0,0 +1,194 @@
+    const dbPath =
+      storagePath !== undefined
+        ? path.join(storagePath, CLIENT_SQLITE_FILE_NAME)
+        : ':memory:'
+
+    const sqlite = new Database(dbPath)
+    this.#db = drizzle(sqlite)
+    migrate(this.#db, { migrationsFolder: './drizzle/client' })
+
+    this.#keyManager = new KeyManager(KeyManager.generateRootKey())
+    this.#projectSettingsIndexWriter = new IndexWriter({
+      tables: [projectTable],
+      sqlite,
+    })
+    this.#storagePath = storagePath
+  }
+
+  /**
+   * Create a new project.
+   * @param {import('type-fest').Simplify<Partial<Omit<import("@mapeo/schema").ProjectValue, 'schemaName'>>>} settings
+   * @returns {Promise<string>}
+   */
+  async createProject(settings) {
+    // 1. Create project keypair
+    const projectKeypair = {
+      publicKey: randomBytes(32),
+      secretKey: randomBytes(64),
+    }
+
+    // 2. Create namespace encryption keys
+    /** @type {Record<import('./core-manager/core-index.js').Namespace, Buffer>} */
+    const namespaceEncryptionKeys = {
+      auth: randomBytes(32),
+      blob: randomBytes(32),
+      blobIndex: randomBytes(32),
+      config: randomBytes(32),
+      data: randomBytes(32),
+    }
+
+    // 3. Save keys to client db in projectKeys table
+    /** @type {ProjectKeys} */
+    const keys = {
+      projectKey: projectKeypair.publicKey,
+      projectSecretKey: projectKeypair.secretKey,
+      authEncryptionKey: namespaceEncryptionKeys.auth,
+      blobEncryptionKey: namespaceEncryptionKeys.blob,
+      blobIndexEncryptionKey: namespaceEncryptionKeys.blobIndex,
+      configEncryptionKey: namespaceEncryptionKeys.config,
+      dataEncryptionKey: namespaceEncryptionKeys.data,
+    }
+
+    // TODO: Update to use @mapeo/crypto when ready (https://github.com/digidem/mapeo-core-next/issues/171)
+    const projectId = projectKeypair.publicKey.toString('hex')
+
+    this.#db
+      .insert(projectKeysTable)
+      .values({
+        projectId,
+        encryptionKeys: this.#keyManager.encryptLocalMessage(
+          serialize(keys),
+          projectId
+        ),
+      })
+      .run()
+
+    // 4. Create MapeoProject instance
+    const project = new MapeoProject({
+      projectSettingsConfig: {
+        db: this.#db,
+        indexWriter: this.#projectSettingsIndexWriter,
+      },
+      storagePath: this.#storagePath,
+      encryptionKeys: namespaceEncryptionKeys,
+      keyManager: this.#keyManager,
+      projectKey: projectKeypair.publicKey,
+      projectSecretKey: projectKeypair.secretKey,
+    })
+
+    // 5. Write project name and any other relevant metadata to project instance
+    await project.$setProjectSettings({
+      // TODO: What's the fallback name?
+      name: settings.name || '',
+      defaultPresets: settings.defaultPresets || {},
+    })
+
+    // 6. Return project id
+    return projectId
+  }
+
+  /**
+   * @param {string} projectId
+   * @returns {MapeoProject}
+   */
+  getProject(projectId) {
```

Comment:
should this be async? nothing called in this requires it to be so

Reply:
In general I've made all methods that access the db async. I did this in case we need to move db queries onto a worker thread in the future - this will make it easier. Plus it means that tests we write here will also work for testing the client implementation (which proxies methods over rpc-reflector).

Reply:
addressed via https://github.com/digidem/mapeo-core-next/pull/199/commits/6d14bacb6f178af5102c8d855efffbb0d3b4c085